cloudinit/sources/DataSourceOracle.py

```diff
@@ -380,75 +392,54 @@ def _add_network_config_from_opc_imds(self, set_primary: bool = False):
-                }
-                self._network_config["config"].append(interface_config)
-            elif self._network_config["version"] == 2:
-                # Why does this elif exist???
-                # Are there plans to switch to v2?
-                interface_config = {
-                    "mtu": MTU,
-                    "match": {"macaddress": mac_address},
-                }
-                self._network_config["ethernets"][name] = interface_config
-
-                interface_config["dhcp6"] = is_primary and is_ipv6_only
-                interface_config["dhcp4"] = is_primary and not is_ipv6_only
-                if not is_primary:
-                    interface_config["addresses"] = []
-                    if vnic_dict.get("privateIp"):
-                        interface_config["addresses"].append(
-                            f"{vnic_dict['privateIp']}/{network.prefixlen}"
-                        )
-                    if vnic_dict.get("ipv6Addresses"):
-                        interface_config["addresses"].append(
-                            f"{vnic_dict['ipv6Addresses'][0]}/"
-                            f"{network.prefixlen}"
-                        )
-                self._network_config["ethernets"][name] = interface_config
+            interface_config = {
+                "mtu": MTU,
+                "match": {"macaddress": mac_address},
+                "set-name": name,
+            }
+            self._network_config["ethernets"][name] = interface_config
+
+            if is_primary and is_ipv6_only:
+                interface_config["dhcp6"] = True
+            elif is_primary and not is_ipv6_only:
+                interface_config["dhcp4"] = True
+
+            if not is_primary:
+                interface_config["addresses"] = []
+                if vnic_dict.get("privateIp"):
+                    interface_config["addresses"].append(
+                        f"{vnic_dict['privateIp']}/{network.prefixlen}"
+                    )
+                if vnic_dict.get("ipv6Addresses"):
+                    interface_config["addresses"].append(
+                        f"{vnic_dict['ipv6Addresses'][0]}/"
+                        f"{network.prefixlen}"
+                    )
+            self._network_config["ethernets"][name] = interface_config
 
 
 class DataSourceOracleNet(DataSourceOracle):
     perform_dhcp_setup = False
 
 
-def _is_ipv4_metadata_url(metadata_address: str):
-    if not metadata_address:
-        return False
-    return metadata_address.startswith(IPV4_METADATA_ROOT.split("opc")[0])
+def convert_v1_netplan_to_v2(network_config: dict) -> dict:
+    has_base_network_key = "network" in network_config
+    if has_base_network_key:
+        network_config = network_config["network"]
+    v1_network_state = network_state.parse_net_config_data(
+        network_config, renderer=netplan.Renderer  # type: ignore
+    )
+    netplan_v2_yaml_string = netplan.Renderer()._render_content(
```

Comment:
I don't actually think this is safe long term. Since v2 is a subset of netplan, it's possible that we'll render something that is valid netplan but not valid v2. I'm wondering if we might just need a separate klibc->v2 renderer rather than trying to support all of v1->v2.